### DISCUSSION TITLE
Fritz and Nibbler doesnt work properly with leela chess 0 versions higher than v26.1 under windows 7 x64

### DISCUSSION THREAD

**Discussion by Vukica14** [LeelaChessZero/lc0#1834] [General]
Hello to everyone,
I am happy to have Nibbler and leela chess 0 but v29 of leela doesnt fit with Nibbler 2.4.0 and its working independently when I am running benchmark for example from cmd propmt.

Previous versions of leela workining perfect so I assume it must be some incompatibility that started from higher versions of 26.1.

Is there any workaround to help to fix this issue?

Its completelz the same with the fritz 16 where I cant make UCI enginee from leela v29 or any higher versions of 0.26.1.

Many thanks in advance.
![2023-01-14 16_29_11-Nibbler](https://user-images.githubusercontent.com/93468705/212479849-20819620-9f27-4c15-84c3-d0494d4a2937.jpg)

**Comment by borg323** [comment]
If I understand correctly the benchmark run is with v0.29.0, so the issue is only about lc0 working with nibbler. We will need more details on how it fails, this is easier done in the #help channel of our chat at <http://lc0.org/chat>.

**Reply by mooskagh** [reply]
I'm not sure whether this is resolved in chat, but if not, one of the reasons may be that it's an opencl backend that runs some tests on the first start and therefore requires some waiting.

**Reply by Vukica14** [reply]
> If I understand correctly the benchmark run is with v0.29.0, so the issue is only about lc0 working with nibbler. We will need more details on how it fails, this is easier done in the #help channel of our chat at http://lc0.org/chat.

Yes Leela works independently, benchmark or "go infinite" with CudNN leela version which I am using exclusively.

I hope now youll have better overview and hopefully some solution for me.

**Comment by Vukica14** [comment]
I spotted right now your responses.
Message I am getting is: Awaiting UCIOK

![2023-01-14 17_22_36-Nibbler](https://user-images.githubusercontent.com/93468705/213037898-61653bfe-90ae-49a6-9fa1-841286705121.jpg)
![2023-01-14 17_08_07-Nibbler](https://user-images.githubusercontent.com/93468705/213037917-4bc3fc19-1b3b-4685-a95b-e234206be7fc.jpg)
![2023-01-14 17_06_43-Nibbler](https://user-images.githubusercontent.com/93468705/213037928-d9e71e6f-38cb-44f2-b46d-3b475984015c.jpg)
![2023-01-14 16_29_11-Nibbler](https://user-images.githubusercontent.com/93468705/213037930-4952a9af-7313-4717-81e9-ca5e7a8b4fb9.jpg)
![2023-01-14 17_29_51-System](https://user-images.githubusercontent.com/93468705/213037858-eacc75eb-0bf7-4b8f-9b07-fd24644aad15.jpg)
![2023-01-14 17_23_55-Nibbler](https://user-images.githubusercontent.com/93468705/213037869-4abf4a50-db49-4587-828d-40b22a770041.jpg)
![2023-01-14 17_43_24-Nibbler](https://user-images.githubusercontent.com/93468705/213037833-1e4f00d5-8b51-4252-a445-aafe78c26405.jpg)

**Comment by Vukica14** [comment]
I posted all screenshoots I made in the Nibbler forum. Nibbler working well with Stockfish and with Leela up to v. 0.26.1. After that Nibbler nor Fritz doesnt work with Leela.

**Comment by borg323** [comment]
I see in the nibbler issue your GPU only has 2Gb of vram - this may be an issue. Did you try to replace the network file that comes with v0.29.0 with the one from v0.26.1 that worked? If that doesn't work, you can try to add `custom_winograd=false` to the backend options.

**Reply by Vukica14** [reply]
> custom_winograd=false

Hi borg, I try but doesn't work with other network file. 
I didn't get other option "custom_winograd=false" where I have to set that?

I searched into enginee and config.json but couldn't find.

I also try in developer window but its the same.

**Reply by borg323** [reply]
You set the `BackendOptions` uci option. No idea how to do that, I don't use nibbler (or any other chess gui).

**Comment by Vukica14** [comment]
Now v0.29 working in Fritz! Only change I can recognize is network I changed as you suggested.

Lets see what is about Nibbler if I can make it work with most recent leela :-D

Here are the parameters of leela under Fritz
![2023-01-18 20_31_35-](https://user-images.githubusercontent.com/93468705/213277164-16b32163-5c21-47d6-ad14-488bf1a4dd0f.jpg)

**Comment by Vukica14** [comment]
Also this is interesting. It seams enginee is not loaded at all. When I try to use other network Ive got this message.
![2023-01-18 20_35_46-Nibbler](https://user-images.githubusercontent.com/93468705/213277590-10bc763d-51e0-4012-9138-558e27952772.jpg)

**Comment by rooklift** [comment]
Uh, if you're detecting the network just by having it in the Lc0 folder, then I guess I'd suggest:

* Find the engines.json
* Close Nibbler
* Delete (or move away) engines.json
* Retry

**Reply by Vukica14** [reply]
> Uh, if you're detecting the network just by having it in the Lc0 folder, then I guess I'd suggest:
> 
> * Find the engines.json
> * Close Nibbler
> * Delete (or move away) engines.json
> * Retry

I try everything but didn't help.

Only I couldnt find what you mention "custom_winograd=false"

I am curious where that can be ?

**Reply by Vukica14** [reply]
> Uh, if you're detecting the network just by having it in the Lc0 folder, then I guess I'd suggest:
> 
> * Find the engines.json
> * Close Nibbler
> * Delete (or move away) engines.json
> * Retry

I try but its the same. Thanks rooklift for all help you provided here. This must be the tricky one. I assume is nothing terrible but its hard to find whats going on.

**Reply by Vukica14** [reply]
> Uh, if you're detecting the network just by having it in the Lc0 folder, then I guess I'd suggest:
> 
> * Find the engines.json
> * Close Nibbler
> * Delete (or move away) engines.json
> * Retry

I was thinking that I answered but obviously didnt. 

I try all this things but doesnt work.  Good is that suddenly leela started to work with Fritz but with nibbler it seams there is no help.

Many thanks for your effort and try to help me in this.

**Comment by rooklift** [comment]
In the options part of the relevant entry in engines.json, you would add:

`"BackendOptions": "custom_winograd=false"`

Make sure the file remains valid JSON, you may or may not need to fix trailing commas depending on where it sits relative to other options.

**Reply by Vukica14** [reply]
> "BackendOptions": "custom_winograd=false"

Thanks for the guide but didnt help. I did it as on the picture but response is the same.
![2023-01-19 20_18_34-C__Users_Mike_AppData_Roaming_Nibbler_engines json - Notepad++  Administrator](https://user-images.githubusercontent.com/93468705/213539627-44119913-7b33-42c3-b428-100e5494a41e.jpg)

**Comment by Vukica14** [comment]
also this giving the same scenario
![2023-01-19 20_21_55-C__Users_Mike_AppData_Roaming_Nibbler_engines json - Notepad++  Administrator](https://user-images.githubusercontent.com/93468705/213540040-ff0642ad-286c-4a16-9b78-c1a4c60d6312.jpg)

**Comment by Vukica14** [comment]
with this, there is no error in loading engine.json but awaiting uciok is still there
![2023-01-19 20_29_21-C__Users_Mike_AppData_Roaming_Nibbler_engines json - Notepad++  Administrator](https://user-images.githubusercontent.com/93468705/213541600-496dc40a-d215-4909-bcf0-646ae592d391.jpg)

**Comment by Vukica14** [comment]
Is there any other command I can apply here in order to try to solve this?

**Comment by rooklift** [comment]
I think you said you were trying to use an older set of weights? Where are they located?

Try closing Nibbler and replacing the top entry with this:

``` JSON
"D:\\_LeelaZero\\LEELA\\lc0-v0.29.0-windows-gpu-nvidia-cudnn\\lc0.exe": {
	"args": [],
	"options": {
		"BackendOptions": "custom_winograd=false",
		"WeightsFile": "path\\to\\weights\\12345.pb.gz"
	},
	"search_nodes": 1000000,
	"search_nodes_special": 10000,
	"limit_by_time": false
},
```

But replacing `path\\to\\weights\\12345.pb.gz` with the actual path, and making sure you use double backslashes.

**Reply by Vukica14** [reply]
> I think you said you were trying to use an older set of weights? Where are they located?
> 
> Try closing Nibbler and replacing the top entry with this:
> 
> ```json
> "D:\\_LeelaZero\\LEELA\\lc0-v0.29.0-windows-gpu-nvidia-cudnn\\lc0.exe": {
> 	"args": [],
> 	"options": {
> 		"BackendOptions": "custom_winograd=false",
> 		"WeightsFile": "path\\to\\weights\\12345.pb.gz"
> 	},
> 	"search_nodes": 1000000,
> 	"search_nodes_special": 10000,
> 	"limit_by_time": false
> },
> ```
> 
> But replacing `path\\to\\weights\\12345.pb.gz` with the actual path, and making sure you use double backslashes.

I discovered that new weight file is not backwards compatible so v0.26.1 cant use new one. I try 3 different weight files for the v0.29 but result is the same.

Also I try to change "search_nodes": 1000000 to null as it is for v.0.26.1 but didnt help.

**Comment by Vukica14** [comment]
{
	"D:\\_LeelaZero\\LEELA\\lc0-v0.29.0-windows-gpu-nvidia-cudnn\\lc0.exe": {
		"args": [],
		"options": {
			"BackendOptions": "custom_winograd=false",
			"WeightsFile": "D:\\_LeelaZero\\LEELA\\lc0-v0.29.0-windows-gpu-nvidia-cudnn\\791556.pb.gz"
this is network file provided with v 0.29
		},
		"search_nodes": null,
		"search_nodes_special": 10000,
		"limit_by_time": false
	},
	"D:\\_LeelaZero\\LEELA\\lc0-v0.26.1-windows-gpu-nvidia-cuda\\lc0.exe": {
		"args": [],
		"options": {
			"WeightsFile": "D:\\_LeelaZero\\LEELA\\lc0-v0.26.1-windows-gpu-nvidia-cuda\\703350.pb.gz"
this is network file provided with v 0.26.1
		},
		"search_nodes": null,
		"search_nodes_special": 10000,
		"limit_by_time": false
	},
	"D:\\_LeelaZero\\LEELA\\stockfish_15.1_win_x64_popcnt\\stockfish-windows-2022-x86-64-modern.exe": {
		"args": [],
		"options": {
			"MultiPV": 3
		},
		"search_nodes": null,
		"search_nodes_special": 10000000,
		"limit_by_time": false
	}
}

**Comment by rooklift** [comment]
Yes but I thought you were trying to make 0.29.0 work with the weights 703350?

**Reply by Vukica14** [reply]
> Yes but I thought you were trying to make 0.29.0 work with the weights 703350?

Yes I try and I dont know how in the Fritz leela v0.29 now working but not yet in the Nibbler.

I try 3 different network files with v0.29 but didnt help.

Again what could cause "awaiting uciok" ? Do you have any other idea?

**Comment by rooklift** [comment]
Leela not starting or crashing, and no.

(To be clear: when you say you've tried 3 different weights, do you mean by editing engines.json, or by using the menu in Nibbler? We've established the latter won't work, so you'd have to try the former.)

**Reply by Vukica14** [reply]
![2023-01-20 08_30_04-Lc0 v0 29 0](https://user-images.githubusercontent.com/93468705/213641144-3cf6095a-c70e-42a8-99ef-5defe037afe2.jpg)
> Leela not starting or crashing, and no.
> 
> (To be clear: when you say you've tried 3 different weights, do you mean by editing engines.json, or by using the menu in Nibbler? We've established the latter won't work, so you'd have to try the former.)

I try recently by engine.json as you suggested and previously by using menu. I try with network provided within the package 0.29 than with the one provided within the 0.26.1 package and 3rd try was with one other network file I downloaded in the past and used for most leelas I used but effect was the same in all those cases.
Now I discovered bit more about leela under Fritz that started to work suddenly... Its set as auto-discover and most likely it took provided network as I copied entire leela into Fritz folder enginees.

**Reply by borg323** [reply]
Yes, autodiscover is the default, and it will choose the newest network (checks the date). To make sure, just keep one network in the same directory with lc0 (and make sure there is no "Networks" directory there, as it will be checked as well).

**Reply by rooklift** [reply]
If after:

* Removing all but the small network from the Lc0 folder
* Removing the WeightsFile entry from engines.json (ensuring it remains valid JSON).

... it still doesn't work, I've nothing left to suggest.
